redis.conf

```diff
@@ -1442,6 +1438,9 @@ aof-use-rdb-preamble yes
 # the AOF format in a way that may not be compatible with existing AOF parsers.
 aof-timestamp-enabled no
 
+# If enable we will automatically deleting HISTORY type AOFs. default enabled.
+aof-enable-auto-gc
```

Comment:
At present, I only use `aof-enable-auto-gc` in the test to assert that expected history AOF will be generated in AOFRW.

However, I am wondering whether there is such a scenario: the user has a backup system that will continuously upload all incremental AOFs, because our AOFs have timestamp annotations, so we can even recovery the data to each previous point in time. Therefore, after each AOFRW, we cannot delete the history AOF immediately, unless the user has actually uploaded these AOFs . This is just a guess of mine. I don't mind that it is only used for testing purposes like `aof-child-rewrite-delay` and not exposed in `redis.conf`.

Reply:
ok, i have a feeling that such an external mechanism will require more assistance from redis.
at the very least it be able to trigger GC manually, or clean the manifest in some way (before/after deleting the files).

I guess we should leave this feature out of the scope for now until we design it properly.
so in that case, let's mark this flag as a testing one (comment), and remove the documentation from redis.conf.

i also think it's still a good idea to negate it.

p.s. we do plan to soon add some config.c flag on all test testing configs see https://github.com/redis/redis/issues/9684
i.e. it'll probably be the ones that are not documented in redis.conf.